data_steward/tools/create_tier.py

```diff
@@ -0,0 +1,126 @@
+            f"Parameter ERROR {release_tag} is in an incorrect format, accepted: YYYYq#r#"
+        )
+
+
+def validate_tier_param(tier):
+    """
+    helper function to validate the tier parameter passed is either 'controlled' or 'registered'
+
+    :param tier: tier parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if tier.lower() not in ['controlled', 'registered']:
+        LOGGER.error(
+            f"Parameter ERROR: {tier} is an incorrect input for the tier parameter, accepted: controlled or "
+            f"registered")
+
+
+def validate_deid_stage_param(deid_stage):
+    """
+    helper function to validate the deid_stage parameter passed is correct, must be 'deid', 'base' or 'clean'
+
+    :param deid_stage: deid_stage parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if deid_stage not in ['deid', 'base', 'clean']:
+        LOGGER.error(
+            f"Parameter ERROR: {deid_stage} is an incorrect input for the deid_stage parameter, accepted: "
+            f"deid, base, clean")
+
+
+def create_tier(credentials_filepath, project_id, tier, input_dataset,
+                release_tag, deid_stage):
+    """
+    This function is hte main entry point for the deid process.
+    It passes the required parameters to the implementing functions.
+
+    :param credentials_filepath: filepath to credentials to access GCP
+    :param project_id: project_id associated with the input dataset
+    :param tier: controlled or registered tier intended for the output dataset
+    :param input_dataset: name of the input dataset
+    :param release_tag: release tag for dataset in the format of YYYYq#r#
+    :param deid_stage: deid stage (deid, base or clean)
+    :return: name of created controlled or registered dataset
+    """
+
+    # validation of params
+    validate_release_tag_param(release_tag)
+    validate_tier_param(tier)
+    validate_deid_stage_param(deid_stage)
+
+
+def parse_deid_args(args=None):
+    parser = argparse.ArgumentParser(
+        description='Parse deid command line arguments')
+    parser.add_argument('-c',
+                        '--credentials_filepath',
+                        dest='credentials_filepath',
+                        action='store',
+                        help='file path to credentials for GCP to access BQ',
+                        required=True)
+    parser.add_argument('-p',
+                        '--project_id',
+                        action='store',
+                        dest='project_id',
+                        help=('Project associated with the '
+                              'input dataset.'),
+                        required=True)
+    parser.add_argument('-t',
+                        '--tier',
+                        action='store',
+                        dest='tier',
+                        help='controlled or registered tier',
```

Comment:
I think you should try using the argparse [`type`](https://docs.python.org/3/library/argparse.html#type) or [`choices`](https://docs.python.org/3/library/argparse.html#choices) arguments.

Reply:
Oh neat, I haven't used those before. I will adjust the code and use a user defined function for they type for `release_tag` and either choice or type for the other two. Thanks!

Reply:
Just a quick review of your code does make it look like this PR is covering both DC-1289 and DC-1290.